### PR TITLE
test-case: simultaneous-playback-capture.sh to remove 'both' keyword

### PR DESCRIPTION
**Pull request by Bin-QA**
Need to split both type pipeline into playback and capture
Change both keyword into check duplicate pcm id

Wait patch [sof-test: split both type pipeline into playback and capture](https://github.com/thesofproject/sof-test/pull/80 ) can be merge this PR

Because without that change, this pr couldn't worked

Signed-off-by: Wu, BinX <binx.wu@intel.com>

### CODE REVIEW

**Comment by stripes416** [comment]
Okay, so we need to wait until #80 (which is also dependent on #10) is merged before merging this PR?

File: test-case/simultaneous-playback-capture.sh [hunk 1]
```diff
@@ -37,7 +37,23 @@ tplg=${OPT_VALUE_lst['t']}
 wait_time=${OPT_VALUE_lst['w']}
 loop_cnt=${OPT_VALUE_lst['l']}
 
-func_pipeline_export $tplg "type:both"
+
```
Comment (stripes416) [inline test-case/simultaneous-playback-capture.sh]: lets remove this extra line here

File: test-case/simultaneous-playback-capture.sh [hunk 2]
```diff
@@ -37,7 +37,23 @@ tplg=${OPT_VALUE_lst['t']}
...
+# get 'both' pcm, it means pcm have same id with different type
+declare -A tmp_id_lst
+id_lst_str=""
+for i in $(sof-tplgreader.py -t $TPLG_ROOT $TPLG -d id -v)
+do
+    if [ ! "${tmp_id_lst["$i"]}" ];then  # this id is never used
```
Comment (stripes416) [inline test-case/simultaneous-playback-capture.sh]:
```suggestion
    if [ ! "${tmp_id_lst["$i"]}" ]; then  # this id is never used
```

File: test-case/simultaneous-playback-capture.sh [hunk 3]
```diff
@@ -37,7 +37,23 @@ tplg=${OPT_VALUE_lst['t']}
...
+        tmp_id_lst["$i"]=0
+    else # this id already used
+        tmp_id_lst["$i"]=1
+        id_lst_str="$id_lst_str,$i"
+    fi
+done
+# now all duplicate id already been catched
```
Comment (stripes416) [inline test-case/simultaneous-playback-capture.sh]:
```suggestion
# now all duplicate ids have already been caught
```

File: test-case/simultaneous-playback-capture.sh [hunk 4]
```diff
@@ -54,8 +70,10 @@ do
     dlogi "Testing: (Loop: $i/$loop_cnt)"
     # clean up dmesg
     sudo dmesg -C
-    for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+    # follow sof-tplgreader the capture will after playback
```
Comment (stripes416) [inline test-case/simultaneous-playback-capture.sh]: I'm not sure what you're trying to say here. Can you expand on this a bit?

Comment (Bin-QA) [reply test-case/simultaneous-playback-capture.sh]: @stripes416 check with patch #80 the both will split 2 pipeline the order of those pipeline is 1st is playback, 2nd is capture, so here step change to 2

Comment (stripes416) [reply test-case/simultaneous-playback-capture.sh]: Okay, I think I understand what you're trying to say here now. Maybe change the comment to something like below:

```suggestion
    # following sof-tplgreader, split 'both' pipelines into separate playback & capture pipelines, with playback occurring first
```

**Comment by xiulipan** [comment]
@stripes416 @aiChaoSONG Please help to review and approve as all dependency is merged.

**Comment by aiChaoSONG** [comment]
@Bin-QA still need some fix. hw:0,1, hw:0,2 shouldn't be present
```
2020-02-25 08:26:23 UTC [COMMAND] aplay -D hw:0,0 -c 2 -r 48000 -f S16_LE /dev/zero -q &
2020-02-25 08:26:23 UTC [COMMAND] arecord -D hw:0,0 -c 2 -r 48000 -f S16_LE /dev/null -q &
2020-02-25 08:26:23 UTC [INFO] Preparing to sleep for 5
2020-02-25 08:26:28 UTC [INFO] check pipeline after 5s
2020-02-25 08:26:28 UTC [COMMAND] killing all pipelines
2020-02-25 08:26:28 UTC [COMMAND] aplay -D hw:0,1 -c 4 -r 48000 -f S32_LE /dev/zero -q &
2020-02-25 08:26:28 UTC [COMMAND] arecord -D hw:0,2 -c 2 -r 48000 -f S16_LE /dev/null -q &
aplay: main:788: audio open error: No such file or directory
2020-02-25 08:26:28 UTC [INFO] Preparing to sleep for 5
arecord: main:788: audio open error: No such file or directory
2020-02-25 08:26:33 UTC [INFO] check pipeline after 5s
...
2020-02-25 08:26:35 UTC [INFO] Test Result: FAIL!
```

**Comment by Bin-QA** [comment]
> @Bin-QA still need some fix. hw:0,1, hw:0,2 shouldn't be present

The root cause is by typo, because we current use `func_pipeline_export`
which will help to check tplg file name, here use the error path & tplg, so the filter is not work  
already update to fix it

**Pull request merged**